src/superstream_producer.rs

```diff
@@ -0,0 +1,135 @@
+use crate::error::ProducerCloseError;
+use crate::{
+    client::Client,
+    environment::Environment,
+    error::{ProducerCreateError, ProducerPublishError},
+    producer::{ConfirmationStatus, NoDedup, Producer},
+    superstream::{DefaultSuperStreamMetadata, RoutingStrategy},
+};
+use rabbitmq_stream_protocol::message::Message;
+use std::collections::HashMap;
+use std::future::Future;
+use std::marker::PhantomData;
+use std::sync::Arc;
+
+//type FilterValueExtractor = Arc<dyn Fn(&Message) -> String + 'static + Send + Sync>;
+
+#[derive(Clone)]
+pub struct SuperStreamProducer<T>(
+    Arc<SuperStreamProducerInternal>,
+    HashMap<String, Producer<T>>,
+    DefaultSuperStreamMetadata,
```

Comment:
@wolf4ood I'm not sure if it is more convenient to define this HashMap and DefaultSuperStreamMetadata inside SuperStreamProducerInternal. I tried to do that like:
```
pub struct SuperStreamProducerInternal {
    pub(crate) environment: Environment,
    client: Client,
    // TODO: implement filtering for superstream
    //filter_value_extractor: Option<FilterValueExtractor>,
    routing_strategy: RoutingStrategy,
    super_stream_metadata: DefaultSuperStreamMetadata,
}
```
but then when I access this field in Send to modify it like:
```
routing_strategy.routes(&message, &mut self.0.super_stream_metadata).await
```

I get this error
```
     routing_strategy.routes(&message, &mut self.0.super_stream_metadata).await
   |                                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot borrow as mutable
```

while it's fine if I leave the field inside SuperStreamProducer (I can access it without problem like this code is doing)